src/MCPServer/lib/RPCServer.py

```diff
@@ -54,206 +65,374 @@ class NotFoundError(RPCServerError):
+        is non-dry given ``approve_transfer_by_path_handler`` above.
+
+        [config]
+        name = approvePartialReingest
+        raise_exc = no
+        """
+        sip_uuid = payload.get("sip_uuid")
+        user_id = payload.get("user_id")
+        job = Job.objects.filter(
+            sipuuid=sip_uuid,
+            microservicegroup="Reingest AIP",
+            currentstep=Job.STATUS_AWAITING_DECISION,
+        ).first()
+        if not job:  # No job to be found.
+            raise NotFoundError(
+                'There is no "Reingest AIP" job awaiting a'
+                " decision for SIP {}".format(sip_uuid))
+        not_found = NotFoundError(
+            "Could not find choice for approve AIP reingest")
+        try:
+            chain = self.workflow.get_chain(self.APPROVE_AIP_REINGEST_CHAIN_ID)
+        except KeyError:
+            raise not_found
+        try:
+            choicesAvailableForUnits[job.pk].proceedWithChoice(
+                chain.id, user_id)
+        except IndexError:
+            raise not_found
+
+    def _get_processing_config_fields_handler(self, worker, job):
+        """List processing configuration fields.
+
+        [config]
+        name = getProcessingConfigFields
+        raise_exc = no
+        """
+        return get_processing_fields(self.workflow)
+
+    def _units_statuses_handler(self, worker, job, payload):
+        """Returns the status of units that are of type SIP or Transfer.
+
+        It returns a JSON-encoded objects. Its ``objects`` attribute is an
+        array of objects, each of which represents a single unit. Each unit
+        has a ``jobs`` attribute shoe value is an array of objects, each of
+        which represents a job of the unit.
+
+        [config]
+        name = getUnitsStatuses
+        raise_exc = no
+        """
+        unit_types = {
+            "SIP": (SIP, "unitSIP"),
+            "Transfer": (Transfer, "unitTransfer"),
+        }
+        try:
+            model_attrs = unit_types[payload["type"]]
+            lang = payload["lang"]
+        except KeyError as err:
+            raise UnexpectedPayloadError("Missing parameter: {}".format(err))
+        model = model_attrs[0]
+        sql = """
+        SELECT SIPUUID,
+               MAX(UNIX_TIMESTAMP(createdTime) + createdTimeDec) AS timestamp
+            FROM Jobs
+            WHERE unitType=%s AND NOT SIPUUID LIKE '%%None%%'
+            GROUP BY SIPUUID;"""
+        with connection.cursor() as cursor:
+            cursor.execute(sql, (model_attrs[1],))
+            sipuuids_and_timestamps = cursor.fetchall()
+        # This is a shared data structure, read safely.
+        with choicesAvailableForUnitsLock:
+            jobs_awaiting_for_approval = choicesAvailableForUnits.copy()
+        objects = []
+        for unit_id, timestamp in sipuuids_and_timestamps:
+            item = {
+                # TODO: consolidate this!
```

Comment:
Are `TODO` comments like this just transplanted from the  Dashboard? We can probably just remove them. This and `# Check if hidden (TODO: this method is slow)` don't really add too much.

Reply:
```python
# TODO: consolidate this!
"id": unit_id, "uuid": unit_id,
```

```python
# Check if hidden (TODO: this method is slow)
if model.objects.is_hidden(unit_id):
    continue
```

Ok, comments removed.